bin/cv_regress

```diff
@@ -155,7 +155,7 @@ def read_file(args, file):
         # Determine if a test is valid, skip for compliance tests
         # Since it is not possible to determine apriori if a compliance test is valid
         if not 'compliance' in test.cmd:
-            check_valid_test(args.project, test.name)
+            check_valid_test(args.project, test.testname)
```

Comment:
This _looks_ like it could break `cv_regress` for other simulators.

Reply:
Yes, but it only _looks_, don't be affraid. 


I also updated in consequence the constructor of the class test inside the file lib/cv_regression.py :

```
class Test:
        ...
        if not hasattr(self, 'testname'):
            self.testname = self.name
        ...
```

By doing so, the testname field is filled with the YAML top item, and backward compatibility is ensured 😄 

Now, a little explanation and more context on why it was necessary. I wanted to be able to run within the same regression a unique test with different core configuration. One practical example is for floating point tests, as we are going to have multiple values for latencies, but the same testcode underlying. With the already existing cv_regress YAML architecture, it was not possible because it would imply having twice the same entry with the same name, and would look like this : 

```
builds:
  uvmt_cv32e40p:
    cmd: make comp
    dir: cv32e40p/sim/uvmt

  uvmt_cv32e40p_pulp:
    ... (same as above) ... 
    cfg: pulp

tests:
  hello-world:    <- issue here
    ...
    build: uvmt_cv32e40p
    cmd: make test COREV=YES TEST=hello-world
    ...

  hello-world:    <- issue here
    ... 
    build: uvmt_cv32e40p_pulp
    cmd: make test COREV=YES TEST=hello-world
```

As you can see inside the above snippet, the already existing cv_regress function you mentionned **forces** to have the "real" test name as the YAML top item as it raises an error if not, and this leads to obvious issues while parsing the YAML file.  

So I added an optionnal field "testname" to the YAML parsing so that I am able to do what I wanted, for example like this : 


```
tests:
  hello-world:    
    ...
    build: uvmt_cv32e40p
    cmd: make test COREV=YES TEST=hello-world
    ...
    ^  ^  ^  
    |  |  | 
    same as before, untouched, without testname

  hello-world-pulp:            <- issue solved
    ... 
    build: uvmt_cv32e40p_pulp
    cmd: make test COREV=YES TEST=hello-world
    testname: hello-world      <- issue solved
```